### PR TITLE
Fix/text input internal state maintained removed

### PR DESCRIPTION
**Pull request by AditiTiwariRazorpay**
Following are read directly from props
1. Input value
2. onChange

### CODE REVIEW

File: packages/blade-old/src/atoms/TextInput/TextInput.native.js [hunk 1]
```diff
@@ -395,7 +375,7 @@ const TextInput = React.forwardRef(
                                   placeholderTextColor={placeholderTextColor}
                                   onFocus={onTextInputFocus}
                                   onBlur={onTextInputBlur}
-                                  onChangeText={onChangeText}
+                                  onChangeText={onChange}
```
Comment (pritishvaidya) [inline packages/blade-old/src/atoms/TextInput/TextInput.native.js:378]: `onChange` and `onChangeText` are two different functions , one contains nativeEvent properties and shouldn't be mergd together even in blade

File: packages/blade-old/src/atoms/TextInput/TextInput.native.js [hunk 2]
```diff
@@ -405,7 +385,7 @@ const TextInput = React.forwardRef(
                                   hasLeftIcon={hasLeftIcon}
                                   maxLength={maxLength}
                                   onLayout={onTextInputLayout}
-                                  value={input}
+                                  value={value}
```
Comment (pritishvaidya) [inline packages/blade-old/src/atoms/TextInput/TextInput.native.js:388]: _TextInput is a controlled component, which means the native value will be forced to match this value prop if provided_
This is a breaking change and value would need to be provided for every TextInput in the app.

**Review by pritishvaidya — CHANGES_REQUESTED** [review]

File: packages/blade-old/src/atoms/TextInput/TextInput.native.js (hunk 1 above)
Comment (AditiTiwariRazorpay) [reply packages/blade-old/src/atoms/TextInput/TextInput.native.js:378]: onchange we are getting here is a prop of blade component which works as onChangeText. (this was used in same way earlier as well).

File: packages/blade-old/src/atoms/TextInput/TextInput.native.js (hunk 2 above)
Comment (AditiTiwariRazorpay) [reply packages/blade-old/src/atoms/TextInput/TextInput.native.js:388]: Already did a sanity for all the input once merge will give QA for sanity as well.

In my knowledge in design system we should not control value internally. This should be upto the client. Please do correct me if I am wrong

Comment (kamleshchandnani) [reply packages/blade-old/src/atoms/TextInput/TextInput.native.js:388]: @AditiTiwariRazorpay we need to cross-check with PG mobile devs too, this is a breaking change for them too and we shall provide a migration path for existing consumers before merging this PR

@shubhnik @shridhar52war

**Comment by kamleshchandnani** [comment]
@AditiTiwariRazorpay can you detail out what was the reason of this change in your PR description?

**Comment by kamleshchandnani** [comment]
@AditiTiwariRazorpay @sumit-gupta91 do we still need this or we shall close it and revisit later?

**Comment by kamleshchandnani** [comment]
Closing this based on the discussions on Slack since it has a lot of breaking changes. Kindly, re-open this once we have a full-fledged solution along with the migration path.

**Pull request closed without merging**